moped-editor/src/views/projects/projectView/ToolbarPlugin.js

```diff
@@ -0,0 +1,360 @@
+import { useEffect, useState } from "react";
+
+import {
+  Box,
+  ButtonGroup,
+  Button,
+} from "@mui/material";
+import {
+  Redo,
+  Undo,
+  Highlight,
+  FormatAlignJustify,
+  FormatAlignLeft,
+  FormatAlignRight,
+  FormatAlignCenter,
+  FormatBold,
+  FormatItalic,
+  FormatStrikethrough,
+  FormatUnderlined,
+  FormatListBulleted,
+  FormatListNumbered,
+  Link,
+  FormatClear
+} from "@mui/icons-material";
+
+import { useLexicalComposerContext } from "@lexical/react/LexicalComposerContext";
+import {
+  $getSelection,
+  $isRangeSelection,
+  CAN_REDO_COMMAND,
+  CAN_UNDO_COMMAND,
+  FORMAT_ELEMENT_COMMAND,
+  FORMAT_TEXT_COMMAND,
+  REDO_COMMAND,
+  SELECTION_CHANGE_COMMAND,
+  UNDO_COMMAND,
+  COMMAND_PRIORITY_LOW
+} from "lexical";
+import { mergeRegister } from "@lexical/utils";
+import { INSERT_ORDERED_LIST_COMMAND, INSERT_UNORDERED_LIST_COMMAND, REMOVE_LIST_COMMAND, $isListNode, $isListItemNode } from "@lexical/list";
+import { TOGGLE_LINK_COMMAND, $isLinkNode } from "@lexical/link";
+
+const richTextAction = {
+  bold: "bold",
+  italics: "italics",
+  underline: "underline",
+  strikethrough: "strikethrough",
+  highlight: "highlight",
+  number: "number",
+  bullet: "bullet",
+  link: "link",
+  left: "left",
+  center: "center",
+  right: "right",
+  justify: "justify",
+  divider: "divider",
+  undo: "undo",
+  redo: "redo",
+  clear: "clear"
+}
+
+const RICH_TEXT_OPTIONS = [
```

Comment:
You had mentioned that you were having issues with the dividers being interactive, and I think it would be totally fine to compose these in the return of the component instead of iterating this options list. Might makes it easier to control of what ends up wrapped in a button group or not.

Reply:
actually found a solution to the dividers issue! i like iterating through the options list so i'll stick with that :)

Reply:
removed those extra options!